### ISSUE TITLE
I have Already Checked Ports 31400 to 31409 Showing "OK FROM PORT 31401-9" using my Computer IP same Response using my External IP Still I Encountered the same Issue " We couldn't reach the required ports on your machine.

### ISSUE THREAD

**Issue by ShinRasetsu** [pi-node/instructions#222]
I have Already Checked Ports 31400 to 31409 Showing "OK FROM PORT 31401-9" using my Computer IP same Response using my External IP Still I Encountered the same Issue " We couldn't reach the required ports on your machine.
Please double-check your configuration."

_Originally posted by @ShinRasetsu in https://github.com/pi-node/instructions/issues/158#issuecomment-1003725429_

**Comment by Malimujo2202** [comment]
Looks like you have to call your ISP !!
If you are under the NAT or if your ISP is blocking ports then you would not be able to configure port forwarding to your local IP. In that case please call your ISP and resolve problem with them.

**Comment by ShinRasetsu** [comment]
> Looks like you have to call your ISP !!
> If you are under the NAT or if your ISP is blocking ports then you would not be able to configure port forwarding to your local IP. In that case please call your ISP and resolve problem with them.

I can check my port status using my external IP... Therefore my ISP is not blocking Port Forwarding.

**Comment by AndyKr0p** [comment]
If this is about random ports shown in Pi Network blocked but some are yet remain opened, I reported the issue a few days ago.

**Comment by ihatejam** [comment]
@ShinRasetsu The built-in open port checker doesn't always work well. Please try a 3rd party port checker like https://www.ipfingerprints.com/portscan.php 
(please don't post screenshots from ipfingerprints or any other 3rd party port checkers without obfuscating your public/external IP address)

If ipfingerprints also show your ports 31400 to 31409 as closed, then you either have an issue with your ISP and/or you need to forward those ports on your local router.

If ipfingerprints shows your ports as open, then you can simply ignore the built-in result and continue past the test and onto the next screen. You will need to leave the portscanner (stellar-dummy container) running until your node is selected to receive the consensus container. This may take a while. In fact I'm unsure if the testnet container will still be pushed out -- but your availability will still be noted!! And it will reportedly be taken into account later (for node activity rewards mentioned in the new Mainnet Mining Formula).

**Comment by ShinRasetsu** [comment]
> @ShinRasetsu The built-in open port checker doesn't always work well. Please try a 3rd party port checker like https://www.ipfingerprints.com/portscan.php 
> (please don't post screenshots from ipfingerprints or any other 3rd party port checkers without obfuscating your public/external IP address)
> 
> If ipfingerprints also show your ports 31400 to 31409 as closed, then you either have an issue with your ISP and/or you need to forward those ports on your local router.
> 
> If ipfingerprints shows your ports as open, then you can simply ignore the built-in result and continue past the test and onto the next screen. You will need to leave the portscanner (stellar-dummy container) running until your node is selected to receive the consensus container. This may take a while. In fact I'm unsure if the testnet container will still be pushed out -- but your availability will still be noted!! And it will reportedly be taken into account later (for node activity rewards mentioned in the new Mainnet Mining Formula). 

Thank you Very much about the information Regarding this Issue....

**Comment by Malimujo2202** [comment]
> > @ShinRasetsu The built-in open port checker doesn't always work well. Please try a 3rd party port checker like https://www.ipfingerprints.com/portscan.php
> > (please don't post screenshots from ipfingerprints or any other 3rd party port checkers without obfuscating your public/external IP address)
> > If ipfingerprints also show your ports 31400 to 31409 as closed, then you either have an issue with your ISP and/or you need to forward those ports on your local router.
> > If ipfingerprints shows your ports as open, then you can simply ignore the built-in result and continue past the test and onto the next screen. You will need to leave the portscanner (stellar-dummy container) running until your node is selected to receive the consensus container. This may take a while. In fact I'm unsure if the testnet container will still be pushed out -- but your availability will still be noted!! And it will reportedly be taken into account later (for node activity rewards mentioned in the new Mainnet Mining Formula).
> 
> Thank you Very much about the information Regarding this Issue....

Yes, because of the problems Pi portchecker has, it is wise to use web port checking, ipfingerprints or pi-mods.de/portchecker
I would like to add to your very correct explanation that in the conditions of having in use consensus container, web tools shows as open just ports 31401, 31402 and 31403.

**Comment by MartinRe63** [comment]
According this info: https://github.com/pi-node/instructions/wiki/(EN)-Ping-request-issue the ICMP must be active within your router (This will enable your router to answer to a ping from outside). Modern Router block this and some of them don't offer the option to activate this. I do see this as an issue --- In general it's not necessary to do the PING - before a check on the TCP/IP Port is done. As a result I think also my node is hanging at "pinging your computer" and also after continue it's not going further. I strongly recommend to remove the ICMP roundtrip from the communication. It will block a mass of persons from running a node and that is excatly that what the pi don't want. By the way ... ipfingerprints - as recommended above - is showing my port's being open. Another aspect: try to ping minepi.com  --- you'll get no response --- why must a pi node response to pure ping requests? Thx.

**Comment by ramu191193** [comment]
Hi Team,
ports are not opened only 31403 and 31404 are showing opened and i tired to add in router but its not working 
![image](https://github.com/pi-node/instructions/assets/8532544/92150fee-ef2d-4f7d-923b-24f13b1b5396)
![image](https://github.com/pi-node/instructions/assets/8532544/32ea45da-e4e9-4a99-8c05-6bdc32b901d2)

any alterative way to add or any suggestions will be helpful

**Comment by cheme75** [comment]
> Hi Team, ports are not opened only 31403 and 31404 are showing opened and i tired to add in router but its not working ![image](https://user-images.githubusercontent.com/8532544/249170318-92150fee-ef2d-4f7d-923b-24f13b1b5396.png) ![image](https://user-images.githubusercontent.com/8532544/249170479-32ea45da-e4e9-4a99-8c05-6bdc32b901d2.png)
> 
> any alterative way to add or any suggestions will be helpful

What exactly do you mean - “it’s not working”. The consensus container is not yet using all ports on my pc, but it’s working - it’s showing the blocks tally in the node app is up to date and the docker stats show the I/O activity, the log shows it’s running.

**Comment by ramu191193** [comment]
> > Hi Team, ports are not opened only 31403 and 31404 are showing opened and i tired to add in router but its not working ![image](https://user-images.githubusercontent.com/8532544/249170318-92150fee-ef2d-4f7d-923b-24f13b1b5396.png) ![image](https://user-images.githubusercontent.com/8532544/249170479-32ea45da-e4e9-4a99-8c05-6bdc32b901d2.png)
> > any alterative way to add or any suggestions will be helpful
> 
> What exactly do you mean - “it’s not working”. The consensus container is not yet using all ports on my pc, but it’s working - it’s showing the blocks tally in the node app is up to date and the docker stats show the I/O activity, the log shows it’s running.

Consensus container is not running its shows absent for couple of months.. for this i think 31401-03 needs to be opened
![image](https://github.com/pi-node/instructions/assets/8532544/4733124d-aacf-436d-bf0a-7cd46d738c20)

**Comment by cheme75** [comment]
You don’t have the consensus container running, the ports checker should run and stop after checking, but it flawed and don’t always work correctly. Try looking at issue #299 and follow the instructions, it may be same problem. Also issue #49 has thread where I was having problems and found fixes. These fixed my node, but it still took it a while to catch up, at least it’s working now. Although I’m not sure we’ll ever see any real life benefit.
